src/newgrf/newgrf_bytereader.h

```diff
@@ -66,35 +65,53 @@ class ByteReader {
 	 */
 	uint32_t ReadDWord()
 	{
-		uint32_t val = this->ReadWord();
-		return val | (this->ReadWord() << 16);
+		auto result = this->consumer.TryReadUint32LE();
+		if (!result.has_value()) throw OTTDByteReaderSignal();
+		return *result;
+	}
+
+	/**
+	 * Read a single DWord (32 bits).
+	 * @note The buffer is NOT advanced.
+	 * @returns Value read from buffer.
+	 */
+	uint32_t PeekDWord()
+	{
+		auto result = this->consumer.PeekUint32LE();
+		if (!result.has_value()) throw OTTDByteReaderSignal();
+		return *result;
 	}
 
-	uint32_t PeekDWord();
 	uint32_t ReadVarSize(uint8_t size);
-	std::string_view ReadString();
+
+	/**
+	 * Read a NUL-terminated string.
+	 * @returns String read from the buffer.
+	 */
+	std::string_view ReadString()
+	{
+		auto result = this->consumer.ReadUntilChar('\0', StringConsumer::KEEP_SEPARATOR);
+		/* Skip the terminating NUL byte, if present. */
+		if (this->consumer.AnyBytesLeft() && !this->consumer.ReadCharIf('\0')) throw OTTDByteReaderSignal();
```

Comment:
Under what circumstance can `ReadCharIf` return `false` in this situation? (Thus triggering the exception)
Won't `ReadUntilChar` end when either there are no bytes left (i.e. `ReadCharIf` is not called), or when it reaches `'\0'` (i.e. `ReadCharIf('\0')` returns `true`).

Reply:
Ah, true. I had an earlier version, where the termination was required. But that was wrong :)